platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ImportMigrationConfigurationAdminContext.java

```diff
@@ -0,0 +1,315 @@
+
+    private final ConfigurationAdminMigratable admin;
+
+    private final ConfigurationAdmin configurationAdmin;
+
+    /**
+     * Keeps track of all managed services in memory that were not found in the export file so
+     * we know what to delete at the end.
+     */
+    private final Map<String, Configuration> memoryServices;
+
+    /**
+     * Keeps track of all managed service factories in memory that were not found in the export file
+     * so we know what to delete at the end.
+     */
+    private final Map<String, List<Configuration>> memoryFactoryServices;
+
+    private final Map<String, ImportMigrationConfigurationAdminEntry> exportedServices;
+
+    private final Map<String, List<ImportMigrationConfigurationAdminEntry>> exportedFactoryServices;
+
+    private boolean isValid = true; // until proven otherwise
+
+    public ImportMigrationConfigurationAdminContext(ImportMigrationContext context,
+            ConfigurationAdminMigratable admin, ConfigurationAdmin configurationAdmin,
+            Configuration[] memoryConfigs) {
+        super(context);
+        Validate.notNull(admin, "invalid null configuration admin migratable");
+        Validate.notNull(configurationAdmin, "invalid null configuration admin");
+        Validate.notNull(memoryConfigs, "invalid null configurations");
+        this.admin = admin;
+        this.configurationAdmin = configurationAdmin;
+        // categorize memory configurations
+        this.memoryServices = Stream.of(memoryConfigs)
+                .filter(ImportMigrationConfigurationAdminContext::isManagedService)
+                .collect(Collectors.toMap(Configuration::getPid, Function.identity()));
+        this.memoryFactoryServices = Stream.of(memoryConfigs)
+                .filter(ImportMigrationConfigurationAdminContext::isManagedServiceFactory)
+                .collect(Collectors.groupingBy(Configuration::getFactoryPid));
+        // categorize exported configurations
+        final ImportMigrationConfigurationAdminEntry[] entries = context.entries()
+                .map(this::proxy)
+                .filter(Objects::nonNull)
+                .toArray(ImportMigrationConfigurationAdminEntry[]::new);
+
+        this.exportedServices = Stream.of(entries)
+                .filter(ImportMigrationConfigurationAdminEntry::isManagedService)
+                .collect(Collectors.toMap(ImportMigrationConfigurationAdminEntry::getPid,
+                        Function.identity()));
+        this.exportedFactoryServices = Stream.of(entries)
+                .filter(ImportMigrationConfigurationAdminEntry::isManagedServiceFactory)
+                .collect(Collectors.groupingBy(ImportMigrationConfigurationAdminEntry::getFactoryPid));
+        context.getReport()
+                .doAfterCompletion(this::deleteUnexportedConfigurationsAfterCompletion);
+    }
+
+    private static boolean isManagedServiceFactory(Configuration cfg) {
+        return cfg.getFactoryPid() != null;
```

Comment:
❗️ Same (but reverse) as in the `isManagedService` method comment.

Reply:
see below